### PR TITLE
[Feature][Connector-V2][TiKV] Support TiKV Source & Sink

### PR DESCRIPTION
**Pull request by Xuxiaotuan**
## Purpose of this pull request

Add new new connecotor of TiKV source and sink,
refer to https://github.com/apache/incubator-seatunnel/issues/1946#issuecomment-1253260625.

## Check list

* [x] Code changed are covered with tests.
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Xuxiaotuan** [comment]
@Hisoka-X PTAL

**Comment by Hisoka-X** [comment]
Can you add doc to introduce this connector? Reference: https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2

**Comment by Hisoka-X** [comment]
And tell the user why implementing in Java client is better than JDBC, or the different between two way.

**Comment by Xuxiaotuan** [comment]
> And tell the user why implementing in Java client is better than JDBC, or the different between two way.

sure 😃

**Comment by Xuxiaotuan** [comment]
@Hisoka-X I have fixed some code style issues, please rerun ci workflow again

**Comment by Hisoka-X** [comment]
You can package in local before you push. So you can find problem eariler.

**Comment by Xuxiaotuan** [comment]
> 

Ok,  thanks, I'm testing locally and there are still some trouble, **so don't review the code first.** 

The simple scenario test passed, but there are still some types that are not compatible, and there are still some problems in requesting tidb in the docker environment. I am dealing with

image

**Review by ashulin — COMMENTED** [review]
Expect the connector to complete.

And can you move e2e IT to seatunnel-connector-v2-e2e module? you can see #2924.

File: .idea/vcs.xml [hunk 1]
```diff
@@ -1,32 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hailin0) [inline .idea/vcs.xml]: revert this file

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-tikv-flink-e2e/src/test/resources/tikv/pd_tikv_contanner_docker_compose.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+version: '2.1'
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-tikv-flink-e2e/src/test/resources/tikv/pd_tikv_contanner_docker_compose.yml]: remove this file

File: seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/sink/TiKVSinkWriter.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        this.clientSession = new ClientSession(tikvParameters);
+        this.seaTunnelRowType = seaTunnelRowType;
+    }
+
+    @Override
+    public void write(SeaTunnelRow seaTunnelRow) {
+        RawKVClient client = clientSession.session.createRawClient();
+        TiKVDataType tikvDataType = tikvParameters.getTikvDataType();
+
+        // todo 需要兼容序列化不同类型的数据 只是key类型
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/sink/TiKVSinkWriter.java]: use english？

File: seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/sink/TiKVSinkWriter.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    public TiKVSinkWriter(SeaTunnelRowType seaTunnelRowType, TiKVParameters tikvParameters) {
+        this.tikvParameters = tikvParameters;
+        this.clientSession = new ClientSession(tikvParameters);
+        this.seaTunnelRowType = seaTunnelRowType;
+    }
+
+    @Override
+    public void write(SeaTunnelRow seaTunnelRow) {
+        RawKVClient client = clientSession.session.createRawClient();
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/sink/TiKVSinkWriter.java]: Can it be reused?

File: seatunnel-connectors-v2/connector-tikv/pom.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-format-json</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-tikv/pom.xml]: remove this  dependency

https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/pom.xml#L77

**Review by hailin0 — COMMENTED** [review]
can you add docs?

reference
https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/source/Redis.md
https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/sink/Redis.md

File: seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/source/TiKVSource.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        return seaTunnelRowType;
+    }
+
+    @Override
+    public AbstractSingleSplitReader<SeaTunnelRow> createReader(SingleSplitReaderContext readerContext) throws Exception {
+        return new TiKVSourceReader(tikvParameters, readerContext, deserializationSchema);
+    }
+
+    @Override
+    public Serializer<SingleSplitEnumeratorState> getEnumeratorStateSerializer() {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/source/TiKVSource.java]: remove this method

**Comment by Carl-Zhou-CN** [comment]
@Xuxiaotuan hi，I am currently completing the e2e of Tidb, but when I use Docker to start the cluster, there is always a problem with the connection between PD and Tidb. Do you know the reason

 [util.rs:549] ["PD failed to respond"] [err="Grpc(RpcFailure(RpcStatus { code: 14-UNAVAILABLE, message: \"failed to connect to all addresses\", details: [] }))"] [endpoints=127.0.0.1:2379]

**Comment by EricJoy2048** [comment]
Please fix the code style.

**Comment by Xuxiaotuan** [comment]
> @Xuxiaotuan hi，I am currently completing the e2e of Tidb, but when I use Docker to start the cluster, there is always a problem with the connection between PD and Tidb. Do you know the reason
> 
> [util.rs:549] ["PD failed to respond"] [err="Grpc(RpcFailure(RpcStatus { code: 14-UNAVAILABLE, message: "failed to connect to all addresses", details: [] }))"] [endpoints=127.0.0.1:2379]

when accessing between tidb pd and tikv, it is accessed through the internal address side, similar to hadoop  xxx:port

File: seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/source/TiKVSource.java (hunk 6 above)
Comment (Xuxiaotuan) [reply seatunnel-connectors-v2/connector-tikv/src/main/java/org/apache/seatunnel/connectors/seatunnel/tikv/source/TiKVSource.java]: > remove this method

ok，I'm going to modify it

**Comment by Xuxiaotuan** [comment]
> > @Xuxiaotuan hi，I am currently completing the e2e of Tidb, but when I use Docker to start the cluster, there is always a problem with the connection between PD and Tidb. Do you know the reason
> > [util.rs:549] ["PD failed to respond"] [err="Grpc(RpcFailure(RpcStatus { code: 14-UNAVAILABLE, message: "failed to connect to all addresses", details: [] }))"] [endpoints=127.0.0.1:2379]
> 
> when accessing between tidb pd and tikv, it is accessed through the internal address side, similar to hadoop xxx:port

https://mp.weixin.qq.com/s/36EzhuTVb6oiUZTUYSJMWg  tidb is currently accessible in docker

**Comment by Xuxiaotuan** [comment]
> 

I checked and found that when start only tikv and pd, docker is required to use [host-network-mode-linux](https://github.com/pingcap/tidb-docker-compose#host-network-mode-linux) Because I am using macOS, I encountered some problems, in fact，[“The host networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.”](https://www.reddit.com/r/docker/comments/uede0r/cannot_access_to_container_with_host_network_mode/)

**Comment by Carl-Zhou-CN** [comment]
> > > @Xuxiaotuan hi，I am currently completing the e2e of Tidb, but when I use Docker to start the cluster, there is always a problem with the connection between PD and Tidb. Do you know the reason
> > > [util.rs:549] ["PD failed to respond"] [err="Grpc(RpcFailure(RpcStatus { code: 14-UNAVAILABLE, message: "failed to connect to all addresses", details: [] }))"] [endpoints=127.0.0.1:2379]
> > 
> > 
> > when accessing between tidb pd and tikv, it is accessed through the internal address side, similar to hadoop xxx:port
> 
> https://mp.weixin.qq.com/s/36EzhuTVb6oiUZTUYSJMWg tidb is currently accessible in docker

Thank you. I'll try that, too

**Comment by Xuxiaotuan** [comment]
transfer to [[Feature][Connector-V2][TiKV] Support TiKV Source & Sink #3173](https://github.com/apache/incubator-seatunnel/pull/3173)

**Pull request closed without merging**